apps/system/test/unit/remote/simple_app_window_manager_test.js

```diff
@@ -0,0 +1,128 @@
+
+  var subject;
+  var container;
+
+  setup(function() {
+    container = document.createElement('div');
+    container.id = "windows";
+    document.body.appendChild(container);
+
+    subject = BaseModule.instantiate('SimpleAppWindowManager');
+    subject.start();
+  });
+
+  teardown(function() {
+    subject.stop();
+    document.body.removeChild(container);
+  });
+
+  suite('start', function() {
+    test('should set the container', function() {
+      assert.isNotNull(subject.container);
+    });
+  });
+
+  suite('stop', function() {
+    test('should clear the container', function() {
+      subject.stop();
+      assert.isNull(subject.container);
+    });
+  });
+
+  suite('launchApp', function() {
+    var fakeElement;
+    var fakeConfig = {
+      url: 'test.html'
+    };
+
+    setup(function() {
+      fakeElement = document.createElement('iframe');
+      this.sinon.stub(window, 'BrowserFrame', function(config) {
+        this.element = fakeElement;
+      });
+      subject.launchApp(fakeConfig);
+    });
+
+    teardown(function() {
+      fakeElement = null;
+    });
+
+    test('should pass untouched config to BrowserFrame', function() {
+      assert.isTrue(window.BrowserFrame.calledWith(fakeConfig));
+      assert.isNotNull(subject.currentApp);
+    });
+
+    test('should create a container for browserFrame', function() {
+      var appWindow = document.querySelector('.appWindow');
+      assert.isNotNull(appWindow);
+      assert.equal(subject.currentApp.container, appWindow);
+    });
+  });
+
+  suite('killCurrentApp', function() {
+    var fakeElement;
+    var fakeConfig = {
+      url: 'test.html'
+    };
+
+    setup(function() {
+      fakeElement = document.createElement('iframe');
+      this.sinon.stub(window, 'BrowserFrame', function(config) {
+        this.element = fakeElement;
+      });
+      subject.launchApp(fakeConfig);
+      subject.killCurrentApp();
+    });
+
+    teardown(function() {
+      fakeElement = null;
+    });
+
+    test('should remove element from container', function() {
+      var appWindow = document.querySelector('.appWindow');
+      assert.isNull(appWindow);
+      assert.isNull(subject.currentApp);
+    });
+  });
+
+  suite('event handlers', function() {
+    test('should kill current app when mozbrowsererror', function() {
+      this.sinon.stub(subject, 'killCurrentApp');
+      subject._handle_mozbrowsererror();
+      assert.isTrue(subject.killCurrentApp.called);
+    });
+
+    test('should kill current app when mozbrowserclose', function() {
+      this.sinon.stub(subject, 'killCurrentApp');
+      subject._handle_mozbrowserclose();
+      assert.isTrue(subject.killCurrentApp.called);
+    });
+
+    test('should remove animation class when animationend', function() {
+      var fakeElement = document.createElement('div');
+      fakeElement.classList.add('opening');
+      assert.isTrue(fakeElement.classList.contains('opening'));
+
+      subject._handle_animationend({
```

Comment:
better: <code>subject.element.dispatchEvent(new CustomEvent('animationend'));</code> to simulate animationend instead of calling handler directly